### DISCUSSION TITLE
questions about github actions and secrets in public repo

### DISCUSSION THREAD

**Discussion by FOSSgeek** [community/community#40231] [New to GitHub]
hi everyone i wish you have a nice day, so I'm new to github and i was asking my self some security questions about how action secrets work in public repo, in short my questions is (some of them i found answers for them but not sure)
suppose i have a public repo and i did restrict and direct write access by making any contribution only possible by pull request:

1. does any new created branch (from some contributor ) in my repo  have access to my actions secrets if i don't merge the new branch to main.

2.  when i go to actions log i can't expand some steps ( deploy step)
![actionsLog](https://user-images.githubusercontent.com/117503511/204163288-211409ee-78f2-41c9-a501-bfd27b6a2f0b.png)

3. when i change the settings in my repo to restrict code changing to only pull request, is the ability for contributor to make new branchs a security threat at all?

4. what are the possible security threat that i should be aware when i have contributors in my public repo and i have actions secrets ?

5. how big public repo like material ui put lots of workflow files with secrets https://github.com/mui/material-ui/tree/master/.github/workflows  on it without trusting the huge like of contributors the repo have 

any answer or info in these topics is appreciated ❤️

**Comment by airtower-luna** [comment]
The key here is this piece [from the documentation on `pull_request` events](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflows-in-forked-repositories):

> With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository. The GITHUB_TOKEN has read-only permissions in forked repositories.

So the key to keeping your secrets safe is *not* to add people as contributors unless you trust them fully. Let them make PRs from their own forks, that way they can't get access to your secrets.

The catch is of course that the `pull_request` workflows won't be able to do anything that needs secrets. If you just want to run tests it shouldn't matter, but if you have something that absolutely needs secrets, see: https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
